### ISSUE TITLE
Training Instructions 

### ISSUE THREAD

**Issue by abhigoku10** [yichen928/GEAL_active_learning#4]
@yichen928  can you please le tme knw how to train the source code on the the following instruction is no clear 
"Usage
Put our config file ssd300_voc0712_al.py into mmdetection/configs/pascal_voc, which includes the same training hyperparameters with prior active learning work.
Customize the training split files with the ones generated from our data_selection code by specifying --cfg-options data.train.dataset.ann_file=."

and not able to find the tools/trian.py file 
Thanks in advance

**Comment by yichen928** [comment]
Hi thank you for your interest. The object detection code comes from [mmdetection](https://github.com/open-mmlab/mmdetection). Please clone this repo to your computer, and put "ssd300_voc0712_al.py" to "mmdetection/configs/pascal_voc".

**Comment by abhigoku10** [comment]
@yichen928  thanks for the response i was able to resolve the issues but currently get output as below 
2022-03-05 11:09:15,927 - mmdet - INFO - Saving checkpoint at 30 epochs
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 4952/4952, 60.9 task/s, elapsed: 81s, ETA:     0s
---------------iou_thr: 0.5---------------
2022-03-05 11:10:41,260 - mmdet - INFO - 
+-------------+------+-------+--------+-------+
| class       | gts  | dets  | recall | ap    |
+-------------+------+-------+--------+-------+
| aeroplane   | 285  | 1022  | 0.768  | 0.666 |
| bicycle     | 337  | 1034  | 0.804  | 0.719 |
| bird        | 459  | 4713  | 0.808  | 0.652 |
| boat        | 263  | 2495  | 0.768  | 0.531 |
| bottle      | 469  | 4322  | 0.652  | 0.384 |
| bus         | 213  | 1114  | 0.873  | 0.673 |
| car         | 1201 | 5976  | 0.886  | 0.755 |
| cat         | 358  | 1271  | 0.899  | 0.799 |
| chair       | 756  | 7239  | 0.716  | 0.434 |
| cow         | 244  | 1495  | 0.922  | 0.680 |
| diningtable | 206  | 1305  | 0.782  | 0.590 |
| dog         | 489  | 2005  | 0.885  | 0.722 |
| horse       | 348  | 1197  | 0.891  | 0.732 |
| motorbike   | 325  | 1302  | 0.852  | 0.723 |
| person      | 4528 | 20707 | 0.849  | 0.695 |
| pottedplant | 480  | 4687  | 0.610  | 0.367 |
| sheep       | 242  | 1483  | 0.785  | 0.608 |
| sofa        | 239  | 1125  | 0.808  | 0.640 |
| train       | 282  | 1325  | 0.858  | 0.741 |
| tvmonitor   | 308  | 2458  | 0.825  | 0.627 |
+-------------+------+-------+--------+-------+
| mAP         |      |       |        | 0.637 |
+-------------+------+-------+--------+-------+
2022-03-05 11:10:41,342 - mmdet - INFO - Exp name: ssd300_voc0712_al.py
2022-03-05 11:10:41,342 - mmdet - INFO - Epoch(val) [30][4952]	mAP: 0.6370, AP50: 0.6370

any idea y i am getting this less resutls

**Comment by yichen928** [comment]
Can you share the complete log?

**Comment by abhigoku10** [comment]
[20220305_071309_ed.txt](https://github.com/yichen928/GEAL_active_learning/files/8192015/20220305_071309_ed.txt)
@yichen928 pls find the complete log in the text file

**Comment by yichen928** [comment]
I am not very sure what happens but it may be the following case. 

Your loss decreases slowly in the beginning. There seems to be a problem in the training of mmdetection that the training may get stuck in the beginning and the final performance would be poor if the sample number is low. This problem occasionally happens for any sampling strategies including random sampling so it is out of the scope of our active learning research. My suggestion is to try to train SSD again using the same selected samples.  It would get improved in most cases.

**Comment by abhigoku10** [comment]
@yichen928 i did the following 
1. i changed the python voc_dense_kmeans.py --selected 4000 and created a new data 
2. trained it with updated data and below is the log file
[20220306_142521_edited.txt](https://github.com/yichen928/GEAL_active_learning/files/8194919/20220306_142521_edited.txt)
 
i set up mmcv=1.0.5 version is it what causing the issue

**Comment by yichen928** [comment]
I think 67.1 mAP is close to our reported results with 4000 images. I am not sure what causes the problem of mmdetection with 3000 images, but it is less likely to happen as the image number increases.
